mesa/Person.py

```diff
@@ -206,11 +215,27 @@ def init_person(self): # person command
         # limit education by age
         # notice how this deforms a little the initial setup
         self.education_level = self.max_education_level
-        for level in sorted(list(self.m.education_levels.keys()),reverse=True):
+        for level in sorted(list(self.m.education_levels.keys()), reverse=True):
             max_age = self.m.education_levels.get(level)[1]
             if self.age() <= max_age:
                 self.education_level = level - 1
 
+    def enroll_to_school(self, level):
+        """
+        Given a level of education, this method chooses a school where to enroll the agent
+        and modifies my_school atribute in-place.
+        :param level: int, level of education to enroll
+        """
+        self.potential_school = [school for agent in self.neighbors["household"] for school in agent.my_school if school.education_level == level]
+        if self.potential_school:
+            self.my_school = self.m.rng.choice(self.potential_school)
+        else:
+            self.potential_school = [x for x in self.m.schools if x.diploma_level == level]
+            self.my_school = self.m.rng.choice(self.potential_school)
+        self.my_school.my_students.add(self)
+
```

Comment:
don't we need to add the students links at this point?

Reply:
Links are added later, when all students are already enrolled in schools.

from mesa/mesaPROTON_OC.py (630-634)
```
      for student in school.my_students:
                total_pool = school.my_students.difference({student})
                conn_pool = list(self.rng.choice(list(total_pool), conn, replace=False))
                student.makeSchoolLinks(conn_pool)
```